.github/workflows/ci-build.yml

```diff
@@ -39,104 +48,167 @@ jobs:
 
-    - uses: ammaraskar/gcc-problem-matcher@master
     - name: Build
-      run: cd build && make -j2
+      run: |
+        mkdir build
+        cd build
+
+        echo "::group::CMake"
+        cmake ..
+        echo "::endgroup::"
+
+        echo "::group::Build"
+        echo "Running on $(nproc) cores"
+        make -j$(nproc)
+        echo "::endgroup::"
 
     - name: Test
-      run: cd build && make -j2 test
+      run: |
+        cd build
+        make -j$(nproc) test
 
-  windows:
-    name: Windows
+  macos:
+    name: Mac OS
 
-    runs-on: windows-latest
     strategy:
       fail-fast: false
       matrix:
-        arch: [x86, x64]
+        include:
+        - arch: x64
+          full_arch: x86_64
+
+    runs-on: macos-latest
+    env:
+      MACOSX_DEPLOYMENT_TARGET: 10.9
 
     steps:
     - name: Checkout
       uses: actions/checkout@v2
 
     - name: Prepare vcpkg (with cache)
-      uses: lukka/run-vcpkg@v4
+      uses: lukka/run-vcpkg@v6
       with:
-        vcpkgDirectory: 'c:/vcpkg'
+        vcpkgDirectory: '/usr/local/share/vcpkg'
         doNotUpdateVcpkg: true
-        vcpkgArguments: 'liblzma libpng lzo zlib'
-        vcpkgTriplet: '${{ matrix.arch }}-windows-static'
-
-    - uses: ammaraskar/msvc-problem-matcher@master
-    - name: 'Build'
-      uses: lukka/run-cmake@v3
-      with:
-        cmakeListsOrSettingsJson: CMakeListsTxtBasic
-        useVcpkgToolchainFile: true
-        buildDirectory: '${{ runner.workspace }}/build'
+        vcpkgArguments: 'freetype liblzma lzo'
+        vcpkgTriplet: '${{ matrix.arch }}-osx'
 
     - name: Install OpenGFX
       run: |
-        mkdir -p "C:/Users/Public/Documents/OpenTTD/baseset"
-        cd "C:/Users/Public/Documents/OpenTTD/baseset"
+        mkdir -p ~/Documents/OpenTTD/baseset
+        cd ~/Documents//OpenTTD/baseset
+
+        echo "::group::Download OpenGFX"
         curl -L https://cdn.openttd.org/opengfx-releases/0.6.0/opengfx-0.6.0-all.zip -o opengfx-all.zip
+        echo "::endgroup::"
+
+        echo "::group::Unpack OpenGFX"
         unzip opengfx-all.zip
+        echo "::endgroup::"
+
         rm -f opengfx-all.zip
-      shell: bash
+
+    - name: Install GCC problem matcher
+      uses: ammaraskar/gcc-problem-matcher@master
+
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+
+        echo "::group::CMake"
+        cmake ${GITHUB_WORKSPACE} \
+          -DCMAKE_OSX_ARCHITECTURES=${{ matrix.full_arch }} \
+          -DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-osx \
+          -DCMAKE_TOOLCHAIN_FILE=/usr/local/share/vcpkg/scripts/buildsystems/vcpkg.cmake \
+          # EOF
+        echo "::endgroup::"
+
+        echo "::group::Build"
+        echo "Running on $(sysctl -n hw.logicalcpu) cores"
+        make -j$(sysctl -n hw.logicalcpu)
+        echo "::endgroup::"
 
     - name: Test
       run: |
-        cd ${{ runner.workspace }}/build
-        ctest -C Debug
+        cd build
+        make -j$(sysctl -n hw.logicalcpu) test
```

Comment:
Why the difference between this and `$(nproc)` above?

Why not just use `-j` and let `make` pick the number?

Do tests even run in parallel anyway?

Reply:
Never EVER run `make -j`. It does NOT do what you think :) There are horror stories written about that, but the short version: `make -j` spawns AS MANY PROCESSES AS TASKS. It does NOT limit itself in ANY way :D I repeat: NEVER EVER run `make -j` :D

With that out of the way:

`nproc` is a *nix function, not an OSX. On OSX, the best approximation is getting the amount of logical CPUs.

An excellent write-up about this: https://www.flamingspork.com/blog/2020/11/25/why-you-should-use-nproc-and-not-grep-proc-cpuinfo/

Basically, if we would run OSX in Docker, using this sysctl would be sub-optimal. But as far as I know, there is no real alternative.

Why use this at all instead of `-j2` we had? Simple: OSX runners can run in `-j4`, making them a lot faster :)

Tests in parallel: no clue; it already read `-j2`, so I just replaced that. I just tried it, we do not run it in parallel. But that is a bug in CMake, and we cannot possibly blame the CI for it :) I am fine removing it from the test step, but it is pretty much an auto-response for me to add `-j$(nproc)` when ever I write make :) I leave it up to you if you want it changed :)

Reply:
Ah, I missed that it was linux/osx. Carry on :)